txtorcon/torconfig.py

```diff
@@ -445,9 +445,13 @@ def __init__(self, ports, key_blob_or_type='NEW:BEST', auth=[], ver=2):
         # FIXME nicer than assert, plz
         assert ' ' not in self._key_blob
         assert isinstance(ports, list)
-        if not key_blob_or_type.startswith('NEW:') \
-           and (len(key_blob_or_type) > 825 or len(key_blob_or_type) < 820):
-            raise RuntimeError('Wrong size key-blob')
+        try:
+            key_type, key_blob = key_blob_or_type.split(':', 1)
+        except (AttributeError, ValueError):
+            raise RuntimeError('Wrong key-blob format')
```

Comment:
Yeah, that would kind of be more "if it quacks like a duck" etc. But in this case I think checking for a string explicitly is fine. Also, these days "subclassing builtin types", especially str, isn't really a thing you should do anyway.